src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -495,6 +495,9 @@ def _build_role_scope(resource_group_name, scope, subscription_id):
         if resource_group_name:
             err = 'Resource group "{}" is redundant because scope is supplied'
             raise CLIError(err.format(resource_group_name))
+        from azure.mgmt.core.tools import is_valid_resource_id
+        if scope.startswith('/subscriptions/') and not is_valid_resource_id(scope):
```

Comment:
>if scope.startswith('/subscriptions/') [](start = 8, length = 38)

why do you check `scope.startswith('/subscriptions/')` first, can the scope has other value?

Reply:
yes, in some cases, user only provide `/provides/...` as scope value.